module/sock/posix/posix.c

```diff
@@ -626,8 +627,16 @@ posix_sock_create(const char *ip, int port,
 		return NULL;
 	}
 
-	/* Only enable zero copy for non-loopback sockets. */
-	enable_zcopy_user_opts = opts->zcopy && !sock_is_loopback(fd);
+	enable_zcopy_user_opts = opts->zcopy;
+	if (enable_zcopy_user_opts) {
```

Comment:
if we ever care about this another option could be for the calling code  (ours) to do this check in a separate thread before trying to connect, and then disable opts->zcopy if it's loopback

Reply:
Yes, something like that. We need to ensure async execution of code than can potentially block. That may include getaddrinfo() that is called earlier in this function.